db/schema_tables.cc

```diff
@@ -1095,26 +1120,95 @@ future<> store_column_mapping(distributed<service::storage_proxy>& proxy, schema
 // and or filesystem calls, e.g. fsync.
 constexpr size_t max_concurrent = 8;
 
+// Returns the names of tables affected by a schema mutation.
+// If returns std::nullopt, it means that the set of affected tables is unknown.
+// The mutation must target one of the tables in schema_tables_holding_schema_mutations().
+static
+std::optional<std::set<sstring>> get_affected_tables(const sstring& keyspace_name, const mutation& m) {
```

Comment:
The std::optional is explained, but I think it's too tricky to use std::optional to represent all.

How about std::variant<all_tables, std::unordered_set>? It's verbose but I think verbosity is important for such tricky and critical code.

Reply:
also: why std::set and not std::unordered_set?

Reply:
ACK for both suggestions.